ga4gh/decorators.py

```diff
@@ -0,0 +1,92 @@
+"""
+General purpose decorators
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import functools
+import signal
+import time
+
+
+class TimeoutException(Exception):
+    """
+    A process has taken too long to execute
+    """
+
+
+class Repeat(object):
+    """
+    A decorator to use for repeating a tagged function.
```

Comment:
Right now its only usage is in the test cases:
```
$ git grep Repeat
...
tests/unit/test_decorators.py:    @decorators.Repeat(sleepSeconds)
...
```
But the application I had in mind was waiting for a server to startup.  I have this in another branch:
```
    @utils.Timeout()
    @utils.Repeat()
    def _waitForServerStartup(self):
        self.server.poll()
        if self.server.returncode is not None:
            self._waitForServerErrLines()
            message = "Server process unexpectedly died; stderr: {0}"
            failMessage = message.format(self.getServerErrLines())
            self.fail(failMessage)
        return not self._isServerRunning()
```